src/local_sensitivity/derivative_wrappers.jl

```diff
@@ -222,15 +222,24 @@ function _vecjacobian!(dλ, λ, p, t, S::SensitivityFunction, isautojacvec::Zygo
       vec(f(u, p, t))
     end
     tmp1,tmp2 = back(λ)
-
+    
     if typeof(y) <: ArrayPartition
       dλ .= ArrayPartition(tmp1.x)
     else
       dλ[:] .= vec(tmp1)
     end
-
-    dgrad !== nothing && (dgrad[:] .= vec(tmp2))
     dy !== nothing && (dy[:] .= vec(_dy))
+    if length(dgrad) == length(tmp2)
+      dgrad !== nothing && (dgrad[:] .= vec(tmp2))
+    else
+      for (i, λi) in enumerate(λ)
+        _, back = Zygote.pullback(y, prob.p) do u, p
+          f(u, p, t)[i]
+        end
+        _,tmp2 = back(λi)
```

Comment:
you should be able to just output the whole thing and then pullback lambda transpose. In fact, do you need a separate computation for this? I don't think you do.

Reply:
Do you mean like:
```
_, back = Zygote.pullback(y, plorenz) do u, p
		vec(f(u,p,t))
end
_,tmp2 = back(λ')
```
then I only get one vector, which doesn't seem to agree with the lines above.